toolkit/scripts/analysis.mk

```diff
@@ -86,3 +89,49 @@ sodiff-check: $(BUILT_PACKAGES_FILE) | $(SODIFF_REPO_FILE)
 	<$(BUILT_PACKAGES_FILE) $(SODIFF_SCRIPT) $(RPMS_DIR)/ $(SODIFF_REPO_FILE) $(RELEASE_MAJOR_ID) $(SODIFF_OUTPUT_FOLDER)
 
 package-toolkit: $(SODIFF_REPO_FILE)
+
+######## LICENSE CHECK ########
+
+license_check_build_dir   = $(BUILD_DIR)/license_check_tool
+license_out_dir           = $(OUT_DIR)/license_check
+license_results_file_pkg  = $(license_out_dir)/license_check_results_pkg.json
+license_results_file_imge = $(license_out_dir)/license_check_results_image_$(config_name).json
+
+.PHONY: license-check license-check-img clean-license-check
+
+clean: clean-license-check
+clean-license-check:
+	@echo Verifying no mountpoints present in $(license_check_build_dir)
+	$(SCRIPTS_DIR)/safeunmount.sh "$(license_check_build_dir)" && \
+	rm -rf $(license_check_build_dir) && \
+	rm -rf $(license_out_dir)
+
+# If we are using the default RPMS_DIR as LICENSE_CHECK_DIRS, we are responsible for building the rpms as needed.
+# Can set REBUILD_PACKAGES=n to skip rebuilding the rpms if they are already built.
+ifeq ($(LICENSE_CHECK_DIRS),$(RPMS_DIR))
+license-check: $(LICENSE_CHECK_DIRS)
+endif
+
+##help:target:license-check=Validate all packages in the RPMS_DIR for license compliance. Set LICENSE_CHECK_DIRS to override source.
+license-check: $(go-licensecheck) $(chroot_worker) $(LICENSE_CHECK_EXCEPTION_FILE) $(LICENSE_CHECK_NAME_FILE)
+	$(go-licensecheck) \
+		$(foreach license_dir, $(LICENSE_CHECK_DIRS),--rpm-dirs="$(license_dir)" ) \
+		--exception-file="$(LICENSE_CHECK_EXCEPTION_FILE)" \
+		--name-file="$(LICENSE_CHECK_NAME_FILE)" \
+		--worker-tar="$(chroot_worker)" \
+		--build-dir="$(license_check_build_dir)" \
+		--dist-tag=$(DIST_TAG) \
+		$(if $(filter y,$(LICENSE_CHECK_PEDANTIC)),--pedantic) \
+		--results-file="$(license_results_file_pkg)"
+
+##help:target:license-check-img=Validate all packages needed for an image for license compliance. Must set CONFIG_FILE=<path_to_config>.
+license-check-img: $(go-licensecheck) $(chroot_worker) $(image_package_cache_summary) $(LICENSE_CHECK_EXCEPTION_FILE) $(LICENSE_CHECK_NAME_FILE)
```

Comment:
Is my understanding correct, that the `image_package_cache_summary` dependency is to force the download of the image packages? Also, what happens when you run this for two images one after another without cleaning in the meantime? Will we end up verifying a union of the RPMs in the second run?

Reply:
Correct, and each image build gets its own separate cache. The only pollution you might see is if you run the same image back-to-back after changing something in the config to add/remove packages.